### PR TITLE
FUSETOOLS-2400 - fix broken progress bar for new project wizard

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java [hunk 1]
```diff
@@ -105,12 +105,20 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
...
 		}
 
+		try {
+			Display.getDefault().syncExec( () -> postProjectCreationSetup(prj, subMonitor.split(6)));
+		} finally {
+			subMonitor.done();
+		}
+	}
+
+	private void postProjectCreationSetup(IProject prj, SubMonitor subMonitor) {
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: private void postProjectCreationSetup(IProject prj, IProgressMonitor monitor) {
SubMonitor subMonitor = SubMonitor.convert(monitor, 5)
[...]

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java [hunk 2]
```diff
@@ -105,12 +105,20 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
...
+				template.create(prj, metadata, subMonitor.split(1));
 			} catch (CoreException ex) {
 				ProjectTemplatesActivator.pluginLog().logError("Unable to create project...", ex); //$NON-NLS-1$
 			}
 		}
 
+		try {
+			Display.getDefault().syncExec( () -> postProjectCreationSetup(prj, subMonitor.split(6)));
+		} finally {
+			subMonitor.done();
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: why are you calling done?

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java (hunk 1 above)
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: why 5? shouldn't it be 6?

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java (hunk 2 above)
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: forgot to remove. its obsolete now

**Comment by lhein** [comment]
So on Fedora the behaviour is following...

If I set Fork to true then the progress bar appears right away.
If I set Fork to false then the progress bar is displayed several seconds delayed.

**Comment by lhein** [comment]
testPR

**Comment by bfitzpat** [comment]
Leaving this one unmerged for now.

**Comment by lhein** [comment]
Yeah, leave it unmerged. Aurelien reported that on Windows this causes a whole load of dialogs appear and disappear when clicking finish. We have to investigate that.

**Comment by lhein** [comment]
testPR

**Comment by apupier** [comment]
here are the differences with and without the fork option:
[with fork](https://drive.google.com/open?id=0B1cUBNButvV9aTVEZnlPandwdk0)
[without fork](https://drive.google.com/open?id=0B1cUBNButvV9Vk95Sk11U2U5UzQ)

**Comment by lhein** [comment]
The cause of the "frozen" UI on project creation is not really tackled by this solution of forking the creationRunnable. It causes a lot of noise on windows and that is not acceptable. We should look for a different approach. Closing this PR.

**Pull request closed without merging**